### DISCUSSION TITLE
Network Catalogues/Multiple Users

### DISCUSSION THREAD

**Discussion by thebundok** [Beep6581/RawTherapee#7223] [Ideas]
I would love to see network catalogues and multiple user support integrated into a photo management software. I've been pushing for Lightroom to incorporate this for years, but it would require a full rebuild from the ground up, so it appears there's no appetite for it.

Is that something that would be possible with RawTherapee? Is there any appetite for this kind of development? I know nothing about programming, so I don't know how difficult it is to implement something like this, but this type of collaborative editing functionality (bin locking, project locking, shared edits etc.) already exists in the video editing world. Not sure why it couldn't be implemented for RAW editing.

I would push my employer to abandon Lightroom for the first RAW editor/manager that could implement this.

**Comment by Lawrence37** [comment]
I have not seen any other interest in collaborative editing with RawTherapee or even raw photo development in general. Maybe I am just not aware of it. I don't think RawTherapee would require extensive rewriting to add such a feature. The main roadblock, like with any other feature request, is finding an interested developer.

**Reply by thebundok** [reply]
Thanks for the response. There has [definitely ](https://community.adobe.com/t5/lightroom-classic-discussions/operating-lightroom-cc-classic-via-network-drive/td-p/9997615)been [interest](https://community.adobe.com/t5/lightroom-classic-ideas/p-allow-catalog-to-be-stored-on-a-networked-drive/idc-p/14851162/page/23) from the [Lightroom community](https://community.adobe.com/t5/lightroom-classic-discussions/save-lightroom-catalog-lrcat-on-network-drive-so-others-can-edit/td-p/8341403) for [several years](https://community.adobe.com/t5/lightroom-classic-discussions/lightroom-classic-catalog-on-a-network-drive-onedrive-or-google-drive-school-situation/td-p/10924769).

Many people just want to be able to use a single catalogue between family members, all contributing to a shared library of photos. Many more (like myself) want to be able to have a single shared catalogue for an entire office. 

Yes, multiple catalogs can reference the same group of photos, but the benefit of the shared catalog is in not having to redo work already completed by others. The first RAW editor/photo manager to implement this would undoubtedly draw many more less-than-enthusiastic users away from Lightroom, but may also kick off a race to see who could implement it best.

**Reply by Lawrence37** [reply]
I'm not too familiar with Lightroom terminology, so apologies if I misunderstood. Lightroom catalogs are databases that store the edits, ratings, etc.. RawTherapee does not use a database. Instead, all edits, ratings, etc. are stored in side-car files called processing profiles. These are files that have the extension .pp3. The side-car files can be stored per-user (the "cache"), but by default, the are saved next to the image file. Since all edits are stored beside the images, edits are inherently shared by all users who have read and write access to the containing folder. One can still choose to have per-user edits, or even both. See [Processing Profile Handling](https://rawpedia.rawtherapee.com/Preferences#Processing_Profile_Handling) for more details.

In my first response, for some reason I assumed you were already familiar with how RawTherapee handles edits. I was referring to live collaborative editing (like Google Docs), which is something RawTherapee doesn't have.

**Reply by thebundok** [reply]
Ah! I see! Well that changes things quite dramatically then. I made the incorrect assumption that all RAW management programs utilized a database system similar to Lightroom. In this case, RawTherapee _could_ essentially be exactly what we're looking for. 

So thank you for pointing that out for me.  We're not looking for live collaboration (ala Google Docs) so this pretty much answers my question. :)
